### DISCUSSION TITLE
LaTex tblr sizing the table header row

### DISCUSSION THREAD

**Discussion by russurquhart1** [lvjr/tabularray#487] [Q&A]
I'm using the following LaTeX source to create my table. The table looks fine except for the table header row. The height of the table header row is larger than it has to be. Is there a way I can size that? I don't seem to be putting any sizing information for the table header row.

Thanks for any/all help,

Russ
``` 
\begin{longtblr}[
   caption = {Severity Categories}
]{ 
  rows={ rowsep=9pt }, 
  row{1}={gray!10}, 
  rowhead = 1, %repeat the first row (the header) for each time the table breaks a page
  colspec={
    |Q[c,m,wd=1.0in]|Q[l,m,wd=6.0in]|}
}
  \hline
  \begin{center} \textbf{Severity} \end{center} & %
  \begin{center} \textbf{Description} \end{center} \\ %
  \hline
```

**Comment by muzimuzhi** [comment]
Extra height in row 1 is caused by `center` environments. Horizontal alignment of cells in row 1 can be overwritten by `row{1} = {c}` (or the equivalent `row{1} = {halign=c}`) _after_ `colspec` sets `(halign=)c` and `(halign=)l` for all cells in columns 1 and 2, respectively.

```tex
\documentclass[landscape]{article}
\usepackage{tabularray}
\usepackage{xcolor}

\begin{document}
\begin{longtblr}[
   caption = {Severity Categories}
]{
  %repeat the first row (the header) for each time the table breaks a page
  rowhead = 1,
  colspec = { |Q[c,m,wd=1.0in]|Q[l,m,wd=6.0in]| },
  rows={ rowsep=9pt },
  % set "row{1}" after "colspec" to overwrite "halign"
  row{1}={ gray!10, halign=c, font=\bfseries },
}
  \hline
  Severity & Description \\
  \hline
  a & b
\end{longtblr}
\end{document}
```

![image](https://github.com/lvjr/tabularray/assets/6376638/f8edb37d-8b61-41c4-9883-052fb7a3080f)

BTW please make your example complete and compilable next time.

**Reply by russurquhart1** [reply]
I  changed the source as you suggested, But I'm still having the issue. Here is a compilable source:

```
\documentclass[landscape]{article}
\usepackage{tabularray}
\usepackage{xcolor}
\begin{longtblr}[
   caption = {Silicon Bug Categories}
]{ 
  rowhead = 1, %repeat the first row (the header) for each time the table breaks a page
  colspec={ |Q[c,m,wd=1.0in]|Q[l,m,wd=6.0in]| },
  rows={ rowsep=9pt }, 
  %set "row{1}" AFTER "colspec" to overwrite "halign"
  row{1}={gray!10, halign=c, font=\bfseries }, 
}
  \hline
  Category & \begin{center} Description \end{center} \\ \hline

  \textbf{A} & \textbf{No impact on specifications/software workaround}

Bugs that do not prevent the device from meeting the operational requirements defined in the specification or bugs for
which there are viable software workarounds. \\ \hline
  
  \textbf{B} & \textbf{Specification impact}

 Bugs that prevent features defined in the specification from being implemented properly. \\ \hline
\end{longtblr}
\end{document}

```

**Comment by russurquhart1** [comment]
I think I see what the problem is. The `\begin{center} Description \end{center}` might be causing the problem.  (As you stated in your post, that went over my head.) When I removed those, the header row looks reasonable.
Thanks again for your help and sorry for my dense-ness. :D

**Reply by muzimuzhi** [reply]
Because `center` environment itself adds vertical spacing, no matter within a table cell or outside. This [TeX-SX answer](https://tex.stackexchange.com/a/23653) to question "When should we use `\begin{center}` instead of `\centering`?" gives more info.
